packages/rocketchat-integrations/server/api/api.js

```diff
@@ -0,0 +1,358 @@
+			content_raw: this.request._readableState && this.request._readableState.buffer && this.request._readableState.buffer.toString(),
+			headers: this.request.headers,
+			user: {
+				_id: this.user._id,
+				name: this.user.name,
+				username: this.user.username
+			}
+		};
+		try {
+			const sandboxItems = buildSandbox(compiledScripts[this.integration._id].store);
+			const sandbox = sandboxItems.sandbox;
+			sandbox.script = script;
+			sandbox.request = request;
+			const result = vm.runInNewContext('script.process_incoming_request({ request: request })', sandbox, {
+				timeout: 3000
+			});
+			if (result && result.error) {
+				return RocketChat.API.v1.failure(result.error);
+			}
+			this.bodyParams = result && result.content;
+			logger.incoming.debug('[Process Incoming Request result of Trigger', this.integration.name, ':]');
+			logger.incoming.debug('result', this.bodyParams);
+		} catch ({stack}) {
+			logger.incoming.error('[Error running Script in Trigger', this.integration.name, ':]');
+			logger.incoming.error(this.integration.scriptCompiled.replace(/^/gm, '  '));
+			logger.incoming.error('[Stack:]');
+			logger.incoming.error(stack.replace(/^/gm, '  '));
+			return RocketChat.API.v1.failure('error-running-script');
+		}
+	}
+	if (this.bodyParams == null) {
+		return RocketChat.API.v1.failure('body-empty');
+	}
+	this.bodyParams.bot = {
+		i: this.integration._id
+	};
+	try {
+		const message = processWebhookMessage(this.bodyParams, this.user, defaultValues);
+		if (_.isEmpty(message)) {
+			return RocketChat.API.v1.failure('unknown-error');
+		}
+		return RocketChat.API.v1.success();
+	} catch ({error}) {
+		return RocketChat.API.v1.failure(error);
+	}
+};
+
+const addIntegrationRest = function() {
+	return createIntegration(this.bodyParams, this.user);
+};
+
+const removeIntegrationRest = function() {
+	return removeIntegration(this.bodyParams, this.user);
+};
+
+const integrationSampleRest = function() {
+	logger.incoming.info('Sample Integration');
+	return {
+		statusCode: 200,
+		body: [
+			{
+				token: Random.id(24),
+				channel_id: Random.id(),
+				channel_name: 'general',
+				timestamp: new Date,
+				user_id: Random.id(),
+				user_name: 'rocket.cat',
+				text: 'Sample text 1',
+				trigger_word: 'Sample'
+			}, {
+				token: Random.id(24),
+				channel_id: Random.id(),
+				channel_name: 'general',
+				timestamp: new Date,
+				user_id: Random.id(),
+				user_name: 'rocket.cat',
+				text: 'Sample text 2',
+				trigger_word: 'Sample'
+			}, {
+				token: Random.id(24),
+				channel_id: Random.id(),
+				channel_name: 'general',
+				timestamp: new Date,
+				user_id: Random.id(),
+				user_name: 'rocket.cat',
+				text: 'Sample text 3',
+				trigger_word: 'Sample'
+			}
+		]
+	};
+};
+
+const integrationInfoRest = function() {
+	logger.incoming.info('Info integration');
+	return {
+		statusCode: 200,
+		body: {
+			success: true
+		}
+	};
+};
+
+Api.addRoute(':integrationId/:userId/:token', {
```

Comment:
Can the `{ authRequired: true }` be moved into the same line and not several lines, this way it looks like the api definitions for the api v1.